changelog (#286)
chalk-ai/docs: changelog.mdx

description: Updates to Chalk!
---

## December 23, 2024

### Search and filter logs by query name, log message, deployment, and more!

You can now search logs in the Logs Explorer in the dashboard using a key-value pair query format:

> `key1: value1 key2: value2 ...`. This allows you to combine multiple filters at once. The Logs Explorer

Reviewer: Is there a space between the key value pairs or is it `key1:value1`

Reviewer: This is probably beyond the scope of this change log since we've called in `Logs Explorer` in the dashboard, but datadog calls theirs `Log Explorer`. I wonder if we should do the same.
Like i recognize we are like providing access to multiple different logs but...yah... idk haha

Author: good catch

> displays the available keys for filtering, shown below.

Reviewer: ```suggestion
displays the available keywords for filtering, shown below.
```
? maybe

![Updated log explorer](/img/log-search-filters.png)

### Search by file name in the source code viewer

> In the Source Code viewer, found in the Deployments page of the dashboard, you can now search for files by name

Reviewer 2: ```suggestion
In the Source Code viewer on the Deployments page of the dashboard, you can now search for files by name
```

> in addition to the existing text search through the source code from the linked deployment.

Reviewer: ```suggestion
We've extended our Dashboard's Deployment Source Code Viewer! In addition to searching
for keywords in your source code, you can now also search for your files by name.
```
Maybe a little clearer but don't feel super strongly about this suggestion—feel free to ignore :)

![Source code viewer with file search](/img/view-source-file-explorer.png)

### Include query context in metadata field

> The Chalk Client now offers an [include_metadata](/api-docs#ChalkClient.query.include_meta) option to associate

Reviewer 2: This is wrong. It's ChalkClient.query.query_context, and it's for moving arbitrary key-value pairs into a resolver

Reviewer 2: i need to write docs, will do today
